interp/run.go

```diff
@@ -642,6 +642,15 @@ func assign(n *node) {
 				d(f).SetMapIndex(i(f), s(f))
 				return next
 			}
+		case n.kind == defineStmt:
```

Comment:
Why are we handling only the n.nleft == 1 case (L634) ? i.e. if I slightly change the new test to:

```
package main

func main() {
	foos := []func(){}

	for i := 0; i < 3; i++ {
		a, b := i, i
		_ = a
		foos = append(foos, func() { println(i, b) })
	}
	foos[0]()
	foos[1]()
	foos[2]()
}

// Output:
// 3 0
// 3 1
// 3 2
```

then it still does not work.

Reply:
fixed now, and test case added